### PR TITLE
feat(backend): create site settings graphql endpoints

### LINKED ISSUE #403: Add spatie/laravel-settings & graphql routes as appropriate
### Problem/Why
The application currently has no mechanism for administrators to edit settings from the user interface.  For example, changing the title of the site.

### Goal/What
Provide graphql endpoints for saving and reviewing application settings.

### Solution/How
spatie/larvel-settings is a framework that allows for the storing and retrieval of application settings via repositories such as the database driver.   These settings would be useful for things that an admin not skilled in docker environment setups and configuration could easily edit from the interface.   Many settings such as email SMPT configuration application keys, etc should remain in the .env world for configuration as they should have sensible defaults and only be changed by someone who understands the implications for the application stack as deployed.

### To Do / Acceptance Criteria
- [ ] Install spatie/laravel-settings
- [ ] create a general settings group/migration with a site_title: string setting.
- [ ] create graphql query and mutation for interacting with settings groups
- [ ] create phpunit tests for querying and mutating settings.

### Review Checklist
- [ ]  Unit tests are passing.

### Accessibility Checklist

No accessibility concerns

### PR DESCRIPTION
**Pull request by wreality**
This PR:
* Installs spatie/laravel-settings
* Creates graphql endpoints for site_name
* Implements unit tests for update and retrieval

closes #403

### CODE REVIEW

**Comment by wreality** [comment]
@gmeben or @Sruthin86  this bad boy is ready to review whenever one of you gets the chance

File: backend/config/settings.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        'redis' => [
+            'type' => Spatie\LaravelSettings\SettingsRepositories\RedisSettingsRepository::class,
+            'connection' => null,
+            'prefix' => null,
+        ],
+    ],
+
+    /*
+     * The contents of settings classes can be cached through your application,
+     * settings will be stored within a provided Laravel store and can have an
```
Comment (gmeben) [inline backend/config/settings.php:49]:
```suggestion
     * The contents of settings classes can be cached through your application.
     * Settings will be stored within a provided Laravel store and can have an
```

**Review by gmeben — APPROVED** [review]
Looks good. I'm surprised there were that many style changes in `schema.graphql`.

**Comment by wreality** [comment]
> Looks good. I'm surprised there were that many style changes in `schema.graphql`.

Hmm. I didn't notice that. I should probably revert those formatting changes since they're unrelated to the change.

I don't think we have any linting setup for graphql files so it's probably just my preferences reformatting the file.

**Pull request merged**